File: 2_missing_number/2.py
from typing import List

# https://leetcode.com/problems/contains-duplicate/


class Solution:
    # O(n) time complexity, O(1) space complexity
    def missingNumber(self, nums: List[int]) -> int:
        max = len(nums)
        len_plus_missing = len(nums) + 1
        # Арифм прогрессия. Сумма целых чисел является целым числом
        sum_plus_missing = (0 + max) * len_plus_missing // 2

        return sum_plus_missing - sum(nums)

File: 2_missing_number/2.go
package missing_number

// https://leetcode.com/problems/missing-number/

// O(n) time complexity, O(1) space complexity
func missingNumber(nums []int) int {
	max := len(nums)
	lenPlusMissing := len(nums) + 1
	// Арифм прогрессия. Сумма целых чисел является целым числом
	sumPlusMissing := (0 + max) * lenPlusMissing / 2

	return sumPlusMissing - sum(&nums)
}

func sum(nums *[]int) (sum int) {
	for _, num := range *nums {
		sum += num
	}

	return
}
